libs/langchain-google-cloud-sql-pg/src/vectorStore.ts

```diff
@@ -178,15 +180,86 @@ class PostgresVectorStore extends VectorStore {
+    }
+
+    if (options && !options.metadatas) {
+      options.metadatas = Array.from(documents).map(() => new Object());
+    }
+
+    const tuples = customZip(options?.ids, documents, vectors, options?.metadatas);
+
+    // Insert embeddings
+    for (const [id, content, embedding, metadata] of tuples) {
+      const metadataColNames = this.metadataColumns.length > 0 ? this.metadataColumns.join(",") : "";
+
+      let stmt = `INSERT INTO "${this.schemaName}"."${this.tableName}"(${this.idColumn}, ${this.contentColumn}, ${this.embeddingColumn}, ${metadataColNames}`
+      let values = {
+        id: id,
+        content: content.pageContent,
+        embedding: `[${embedding.toString()}]`
+      }
+      let valuesStmt = " VALUES (:id, :content, :embedding";
+
+      // Add metadata
+      let extra = metadata;
+      for (const metadataColumn of this.metadataColumns) {
+        if (metadata.hasOwnProperty(metadataColumn)) {
+          valuesStmt += `, :${metadataColumn}`
+          values[metadataColumn as keyof typeof values] = metadata[metadataColumn]
+          delete extra[metadataColumn]
+        } else {
+          valuesStmt += " ,null"
+        }
+      }
+
+      // Add JSON column and/or close statement
+      stmt += this.metadataJsonColumn ? `, ${this.metadataJsonColumn})` : ")";
+      if (this.metadataJsonColumn) {
+        valuesStmt += ", :extra)";
+        Object.assign(values, { "extra": JSON.stringify(extra) })
+      } else {
+        valuesStmt += ")"
+      }
+
+      const query = stmt + valuesStmt;
+      await this.engine.pool.raw(query, values)
+    }
+
+    return options?.ids;
+  }
+
   _vectorstoreType(): string {
     return "cloudsqlpostgresql"
   }
-  addVectors(vectors: number[][], documents: DocumentInterface[], options?: { [x: string]: any; }): Promise<string[] | void> {
-    throw new Error("Method not implemented.");
-  }
-  addDocuments(documents: DocumentInterface[], options?: { [x: string]: any; }): Promise<string[] | void> {
-    throw new Error("Method not implemented.");
+
+  /**
+   * Adds documents to the vector store, embedding them first through the
+   * `embeddings` instance.
+   *
+   * @param documents - Array of documents to embed and add.
+   * @param options - Optional configuration for embedding and storing documents.
+   * @returns A promise resolving to an array of document IDs or void, based on implementation.
+   * @abstract
+   */
+  async addDocuments(documents: DocumentInterface[], options?: { [x: string]: any; }): Promise<string[] | void> {
+    let texts = [];
+    let metadatas = [];
+
+    for (const doc of documents) {
+      texts.push(doc.pageContent)
+      metadatas.push(doc.metadata)
+    }
+
+    const ids = options?.hasOwnProperty('ids') ? options.ids : [];
+    const embeddings = await this.embeddings.embedDocuments(texts);
+    const results = await this.addVectors(embeddings, documents, { metadatas, ids });
```

Comment:
You can just pass options through the method
```suggestion
    const embeddings = await this.embeddings.embedDocuments(texts);
    const results = await this.addVectors(embeddings, documents, options);
```

Reply:
Done